js-src/myMoneroApi.js

```diff
@@ -0,0 +1,285 @@
+// @flow
+
+const moneroUtilsPromise = require('../monero_utils/monero_utils.js')
+const nettypeUtils = require('../cryptonote_utils/nettype.js')
+const parserUtils = require('../hostAPI/response_parser_utils.js')
+const sendingFundsUtils = require('../monero_utils/monero_sendingFunds_utils.js')
+const HostedMoneroAPIClient = require('../HostedMoneroAPIClient/HostedMoneroAPIClient.Lite.js')
+const BackgroundResponseParser = require('../HostedMoneroAPIClient/BackgroundResponseParser.web.js')
+
+const MAINNET = nettypeUtils.network_type.MAINNET
+
+export type MyMoneroApiOptions = {
+    appUserAgentProduct: string,
+    appUserAgentVersion: string,
+    apiServer: string,
+    fetch: Function,
+    request: Function,
+    randomBytes: Function,
+    apiKey?: string,
+}
+
+export type MyMoneroWallet = {
+    mnemonic: string,
+    moneroAddress: string,
+    moneroSpendKeyPrivate: string,
+    moneroSpendKeyPublic: string,
+    moneroViewKeyPrivate: string,
+    moneroViewKeyPublic: string
+}
+
+export type BalanceResults = {
+    blockHeight: number,
+    totalReceived: string,
+    totalSent: string
+}
+
+export type QueryParams = {
+    moneroAddress: string,
+    moneroSpendKeyPrivate: string,
+    moneroSpendKeyPublic: string,
+    moneroViewKeyPrivate: string
+}
+
+export type SendFundsParams = QueryParams & {
+    targetAddress: string,
+    floatAmount: number,
+    moneroViewKeyPublic: string,
+    nettype?: string, // 'mainnet' only for now
+    isSweepTx?: boolean,
+    doBroadcast?: boolean,
+    paymentId?: string | null,
+    priority?: number, // 1-4 (Default is 1. Higher # is higher priority and higher fee )
+    onStatus?: (code: number) => void
+}
+
+class MyMoneroApi {
+    options: MyMoneroApiOptions
+    keyImageCache: Object
+    hostedMoneroAPIClientContext: Object
+    hostedMoneroAPIClient: Object
+    moneroUtils: Object
+
+    constructor (options: MyMoneroApiOptions) {
+        this.options = options
+        this.keyImageCache = {}
+        if (options.randomBytes) {
+            if (!global) {
+                global = {}
+            } 
+            if (!global.crypto) {
+                global.crypto = {}
+            }
+            if (!global.crypto.randomBytes) {
```

Comment:
Can be removed as using C++ native random for RN. For browser and nodejs, WASM/ASMJS already call JS crypto libraries for random()